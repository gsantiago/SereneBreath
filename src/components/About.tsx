import React from "react";
import { Popover } from "./Popover";

export function About({ isVisible }) {
  return (
    <Popover isVisible={isVisible} position="bottom">
      <section className="p-3">
        <p>made with love by @gsantiago</p>
      </section>
    </Popover>
  );
}
