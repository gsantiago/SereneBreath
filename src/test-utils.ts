import "vitest-axe/extend-expect";
import "vitest-dom/extend-expect";

export * from "@testing-library/react";
export { axe } from "vitest-axe";
