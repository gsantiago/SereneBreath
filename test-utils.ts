import "vitest-axe/extend-expect";

export * from "@testing-library/react";
export { axe } from "vitest-axe";
