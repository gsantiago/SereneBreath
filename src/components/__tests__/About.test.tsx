import React from "react";
import { test, expect } from "vitest";

import { render, axe } from "../../../test-utils";

import { About } from "../About";

const setup = () => render(<About isVisible />);

test("renders successfuly", () => {
  const { container } = setup();
  expect(container).toMatchSnapshot();
});

test("accessibility", async () => {
  const { container } = setup();
  expect(await axe(container)).toHaveNoViolations();
});
