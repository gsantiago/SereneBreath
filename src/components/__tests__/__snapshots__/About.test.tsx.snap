// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders successfuly 1`] = `
<div>
  <div
    class="mt-3 px-5 text-right text-gray-600"
  >
    <button
      title="About"
      type="button"
    >
      <svg
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.25 11.25l.041-.02a.75.75 0 011.063.852l-.708 2.836a.75.75 0 001.063.853l.041-.021M21 12a9 9 0 11-18 0 9 9 0 0118 0zm-9-3.75h.008v.008H12V8.25z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
